server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -329,10 +334,18 @@ public void close() throws IOException {
 
     protected long getMinReferencedGen() throws IOException {
         assert readLock.isHeldByCurrentThread() || writeLock.isHeldByCurrentThread();
-        long minReferencedGen = Math.min(
+        long uncommitedSeqNo, minReferencedGen;
+        if (this.indexSettings.isRemoteStoreEnabled()) {
```

Comment:
Why do we need to have this condition? If RemoteFsTranslog is being used, doesn't it guarantee that this condition would always be true?

Reply:
can we add an assert here that `indexSetting.isRemoteStoreEnabled()` is true and simplify the code?

Reply:
What are the cases where remote translog could be used independent of the remote segment store.
Can we simplify this by keeping max of the two seq no to retain instead

Reply:
>  Can we simplify this by keeping max of the two seq no to retain instead.

i assume you mean `min` of two seq no . In most of the cases , flush will be very less as compared to the remote refresh sequence number. Hence we end up relying on flushes for remote FS trimming. This increases the number of files in translog by a huge number (til 512 MB is reached).  This in turn increases the recovery time . 

> What are the cases where remote translog could be used independent of the remote segment store.

Currently it is not used independent of remote segment store. But if you think about it , it doesn't depend on remote segment store at all. I am thinking to use remote translog enablement followed by remote segment store enablement for migration of existing indices. So I am not introducing any changes which would need more work later.